File: app/src/main/java/com/cyberflow/sparkle/main/widget/calendar/DateBean.java
package com.cyberflow.sparkle.main.widget.calendar;

import androidx.annotation.NonNull;

public class DateBean {
    int year;
    int month;
    int day;

    boolean isThisMonth = true;

    public DateBean(int year, int month, int day) {
        this.year = year;
        this.month = month;
        this.day = day;
    }

    public int getYear() {
        return year;
    }

    public int getMonth() {
        return month;
    }

    public int getDay() {
        return day;
    }

    public boolean isThisMonth() {
        return isThisMonth;
    }

    public void setThisMonth(boolean isThisMonth) {
        this.isThisMonth = isThisMonth;
    }

    public static String getMonthEngStr(int month) {
        switch (month) {
            case 1:
                return "January";
            case 2:
                return "February";
            case 3:
                return "March";
            case 4:
                return "April";
            case 5:
                return "May";
            case 6:
                return "June";
            case 7:
                return "July";
            case 8:
                return "August";
            case 9:
                return "September";
            case 10:
                return "October";
            case 11:
                return "November";
            case 12:
                return "December";
            default:
                return "";
        }
    }

    @NonNull
    @Override
    public String toString() {
        return year + "-" + month + " : " + day;
    }
}

File: app/src/main/java/com/cyberflow/sparkle/main/widget/calendar/DateBean.kt
package com.cyberflow.sparkle.main.widget.calendar

import androidx.databinding.BaseObservable
import com.cyberflow.sparkle.MyApp
import com.cyberflow.sparkle.R

data class DateBean(
    var year: Int,
    var month: Int = 1,
    var day: Int = 1,
    var isThisMonth: Boolean = true,
): BaseObservable(){
    var checked = false
}

fun getMonthEngStr(month: Int): String? {
    return when (month) {
        1 -> MyApp.instance.getString(R.string.january)
        2 -> MyApp.instance.getString(R.string.february)
        3 -> MyApp.instance.getString(R.string.march)
        4 -> MyApp.instance.getString(R.string.april)
        5 -> MyApp.instance.getString(R.string.may)
        6 -> MyApp.instance.getString(R.string.june)
        7 -> MyApp.instance.getString(R.string.july)
        8 -> MyApp.instance.getString(R.string.august)
        9 -> MyApp.instance.getString(R.string.september)
        10 -> MyApp.instance.getString(R.string.october)
        11 -> MyApp.instance.getString(R.string.november)
        12 -> MyApp.instance.getString(R.string.december)
        else -> ""
    }
}